server/README.md

```diff
@@ -0,0 +1,77 @@
+# API
+<!-- Endpoints are meant to be preceded by /api (/operations == /api/operations)-->
+## Bookings
+- GET `/bookings`
+  - Param: Time frame (start date and end date)
+  - Response: List of all bookings
+  ##### NOTE: This endpoint is intended to be used for statistics 
+
+## Lectures
+- GET `/lectures`
+  - Param: NONE
+  - Response: List of all lectures
+- GET `/lectures/{lecture_id}`
+  - Param: lecture_id
+  - Response: Lecture object for the given id
+- GET `/lectures/{lecture_id}/bookings`
+  - Param: lecture_id
+  - Response: List of all bookings for a given lecture
+- PUT: `/lectures/{lecture_id}`
+  - Param lecture_id
+  - Body: Lecture object (used to possibly cancel the lecture)
+  - Response: NONE
+- POST: `/lectures`
+  - Body: Lecture object (used to add a lecture)
+  - Response: NONE
+- DELETE: `/lectures/{lecture_id}`
+  - Param: lecture_id
+  - Response: NONE
+  
+## Courses
+- GET `/courses`
+  - Param: NONE
+  - Response: List of all courses
+- GET `/courses/{course_id}`
+  - Param: course_id
+  - Response: Course object
+- GET `/courses/{course_id}/lectures`
+  - Param: course_id
+  - Response: List of all lectures for a given course
+- POST `/courses`
+  - Body: Course object
+  - Response: Response status
+  
+## Students
+- GET `/students/{student_id}/courses`
+  - Param: student_id
+  - Response: List of all courses the student is enrolled at
+- GET `/students/{student_id}/bookings`
+  - Param: student_id
+  - Response: List of all the bookings for a given student
+- GET `/students/{student_id}/lectures`
+  - Param: student_id
+  - Query: start_date, end_date
+  - Response: List of all lectures the student is entitled to follow
+  ##### NOTE: This endpoint should receive query parameters for time frame
+- POST `/students`
+  - Body: Student opject
+  - Response: Response status
+- POST `/students/{student_id}/bookings`
+  - Param: student_id
+  - Body: lecture_id
+  - Response: Booking object
```

Comment:
[Story 5] I think we also need a `DELETE` (in the form `/students/{student_id}/bookings/{booking_id}`).

Reply:
Yes, you're right. I probably forgot about it. However, in the db design document, I noticed that the field booking_id is missing, I guess that in order to use this route we should add it

Reply:
Maybe a DELETE `/students/{student_id}/lectures/{lecture_id}` may work, is it too ugly?

Reply:
I also agree with Dario, a booking_id may  simplify many operations

Reply:
In the case of a `booking_id` being present it would not make sense to have it below the `students` endpoint since the `booking_id` would be unique among all the `bookings`.
I would go with `student_id` + `lecture_id` just to don't have to carry another field around to handle bookings, but I can accept DELETE `/bookings/{booking_id}` too.

Reply:
I think that if we have to reflect the DB structure, we should use `/students/{student_id}/lectures/{lecture_id}` though it's not so nice. Otherwise, we should do the same reasoning for the lecture, that can be identified just by course_id and datetime, and this would require to chnage the DB structure.

Reply:
What if we use DELETE `/students/{student_id}/bookings?lecture_id=<lecture_id>` ?

Reply:
I think that, while it may seem ugly, it's a more RESTful approach to have a unique id for a resource instead of using filters. 

> I would go with DELETE `/students/{student_id}/lectures/{lecture_id}` just to don't have to carry another field around to handle bookings, but I can accept DELETE `/bookings/{booking_id}` too.

Yes, we can also avoid the `booking_id` I suppose.